web/src/settings_panel_menu.js

```diff
@@ -151,17 +159,34 @@ export class SettingsPanelMenu {
         this.$main_elem.children("li").removeClass("active");
         this.$curr_li.addClass("active");
 
-        const settings_section_hash = "#" + this.hash_prefix + section;
+        if (section !== "user-list-admin") {
+            const settings_section_hash = "#" + this.hash_prefix + section;
 
-        // It could be that the hash has already been set.
-        browser_history.update_hash_internally_if_required(settings_section_hash);
+            // It could be that the hash has already been set.
+            browser_history.update_hash_internally_if_required(settings_section_hash);
+        }
 
         $(".settings-section").removeClass("show");
 
         settings_sections.load_settings_section(section);
 
         this.get_panel().addClass("show");
 
+        if (section === "user-list-admin") {
+            const hash_components = window.location.hash.slice(1).split(/\//);
+            let key = hash_components[2];
+            const valid_right_side_tab_values = new Set([
+                "active-users-admin",
+                "deactivated-users-admin",
+                "invites-list-admin",
+            ]);
+            if (key === undefined || !valid_right_side_tab_values.has(hash_components[2])) {
+                key = "active-users-admin";
+                const newUrl = "#" + this.hash_prefix + section + "/active-users-admin";
+                history.replaceState({}, "", newUrl);
+            }
+            this.get_valid_users_section_panel(key);
```

Comment:
I think this handling of hash can be done in `hashchange` because the invalid hash can only be used when manually typing urls or coming through a link with incorrect hash. And then this `activate_section_or_default` function can take a parameter like `right_section` which will only be passed if `section` is `user-list-admin`.

Reply:
@sahil839 I have tried to implement it taking examples from `streams` and `user groups` section. Hope this works.